packages/badge/src/FzBadge.vue

```diff
@@ -5,31 +5,55 @@
 </template>
 
 <script setup lang="ts">
-import { computed } from "vue";
+import { computed, useSlots } from "vue";
 
 const props = withDefaults(
   defineProps<{
     /**
      * Color variant
      */
-    color: "black" | "blue" | "error" | "warning" | "success" | "info";
+    color: "black" | "blue" | "error" | "warning" | "success" | "info" | "light" | "dark";
+    /**
+     * The badge size
+     */
+    size: "sm" | "md" | "lg";
   }>(),
   {
     color: "black",
+    size: "md",
   },
 );
 
+const slots = useSlots();
+const label = computed(() => slots.default?.()[0].children?.toString());
+const variant = computed(() => label.value && label.value.length === 1 ? "rounded" : "default");
```

Comment:
I'd have simply applied one or the other class based on this condition in `classes`